Binary files mouseconfig-3.9/PNP-probing/lib/.pnp_probe_com.c.swp and mouseconfig-3.9.cnc/PNP-probing/lib/.pnp_probe_com.c.swp differ
diff -Nur mouseconfig-3.9/PNP-probing/lib/pnp_probe_com.c mouseconfig-3.9.cnc/PNP-probing/lib/pnp_probe_com.c
--- mouseconfig-3.9/PNP-probing/lib/pnp_probe_com.c	Fri Apr  9 12:15:44 1999
+++ mouseconfig-3.9.cnc/PNP-probing/lib/pnp_probe_com.c	Sat Jun 19 20:32:00 1999
@@ -862,9 +862,19 @@
     /* see if we found any PnP signature */
     if (pnp_strlen != 0) {
       if (*pnp_string == 'M') { /* Legacy mouse */
-	    *t = devprobe_create();
- 	    devprobe_setinfo(*t, DEVPROBE_LEGACY);
-	    devprobe_insert(*t, "CLASS", "MOUSE");
+	    /* Legacy ??? - I have a PNP Genius Mouse that falls here */
+	    /* so for sure ... - cavassin */
+	    if (parse_pnp_string(pnp_string, pnp_strlen, &pnp_id)<0) {
+	      *t = devprobe_create();
+ 	      devprobe_setinfo(*t, DEVPROBE_LEGACY);
+	      devprobe_insert(*t, "CLASS", "MOUSE");
+		} else {
+	      *t = make_devprobe(pnp_id);
+	      if (!devprobe_match(*t, "CLASS")) {
+		    /* try to figure out class from PNP fields */
+		    devprobe_guess_pnp_class(*t);
+	      }
+        }
 	    devprobe_insert(*t, "LINUX DEVICE", port);
 	    devprobe_insert(*t, "BUS", "SERIAL");
 
